### ISSUE TITLE
No Metrics for Microservice OAuth2 (Okta) Application in Registry

### ISSUE THREAD

**Issue by stevemac79** [jhipster/jhipster-registry#437]
##### **Overview of the issue**
I have a Microservice application which is configured with OAuth2 / Okta. This was generated with JHipster v6.4.1.
I am running the latest JHipster Registry (v6.2.0), also configured with OAuth2 / Okta.
I can log in to both applications and also use the Swagger UI on the Registry to run the Microservice APIs. However, when I go to Administration > Metrics, the page does not fetch any metrics for the Microservice application.
Looking at the Chrome Dev Tools I can see:

{
  "type" : "https://www.jhipster.tech/problem/problem-with-message",
  "title" : "Forbidden",
  "status" : 403,
  "detail" : "Access is denied",
  "path" : "/management/jhimetrics",
  "message" : "error.http.403"
}

Relevant sections from my application.yml look like:

![image](https://user-images.githubusercontent.com/58238355/82020007-9aed6c80-96cb-11ea-8257-c886fce857c2.png)

.yo-rc.json:
`
{
  "generator-jhipster": {
    "databaseType": "sql",
    "devDatabaseType": "oracle",
    "enableHibernateCache": true,
    "enableSwaggerCodegen": false,
    "enableTranslation": true,
    "jhiPrefix": "jhi",
    "languages": [
      "en",
      "fr"
    ],
    "messageBroker": false,
    "nativeLanguage": "en",
    "packageName": "com.treasury.prime.ms",
    "packageFolder": "com/treasury/prime/ms",
    "prodDatabaseType": "oracle",
    "searchEngine": false,
    "serviceDiscoveryType": "eureka",
    "skipClient": true,
    "testFrameworks": [],
    "websocket": false,
    "baseName": "primems",
    "applicationType": "microservice",
    "authenticationType": "oauth2",
    "serverPort": "8081",
    "jhipsterVersion": "6.4.1",
    "buildTool": "maven",
    "skipUserManagement": true,
    "clientPackageManager": "npm",
    "cacheProvider": "hazelcast",
    "embeddableLaunchScript": false,
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "blueprints": []
  },
  "entities": [
    "Announcement",
    "MdReference",
    "GovernmentEntity",
    "CapitalProject",
    "CapitalProjectFin"
  ]
}

`

I also get a 403 when attempting to access

management/configprops
management/env
management/jhimetrics
management/logfile
management/loggers
management/threaddump

The following work OK:

management/health
management/info

##### **Motivation for or Use Case**

I would like to be able to view metrics and logs for the Microservice app.

##### **Reproduce the error**

Generate a MS application using 6.4.1, configure Okta
Boot up Registry 6.2.0 using Okta config
Navigate to the Metrics page
Observe 403

##### **JHipster Registry Version(s)**

Registry version 6.2.0

##### **Browsers and Operating System**

Client OS: Windows 10
Browsers: Chrome, latest version

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by stevemac79** [comment]
Turned out the issue here was that the "groups" claim configured within Okta (of type "id") was not being picked up by the application, meaning that the getRolesFromClaims method in SecurityUtils was always returning a 0 length array.

FYI -
I followed the instructions for Okta integration listed here: https://developer.okta.com/blog/2018/03/01/develop-microservices-jhipster-oauth

I got it working by doing the following:

- Adding a new claim in the Okta Authorization Server "groupsAT" with Include in Type "access" (in other respects a duplicate of the original "groups" claim
- Updating the code in SecurityUtils:
`    private static Collection<String> getRolesFromClaims(Map<String, Object> claims) {
        return (Collection<String>) claims.getOrDefault("groupsAT",
            claims.getOrDefault("groups", new ArrayList<>()));
    }`
- Upgrading to version 6.2.0 of the JHipster Registry

I haven't fully got to the bottom of it, but it looks like the access claim is the only thing that can be extracted from the JWT token for the microservice application. @mraible - perhaps the instructions on the Okta setup blog should be updated?
